rust_src/src/windows.rs

```diff
@@ -102,7 +102,7 @@ impl LispWindowRef {
         let qfloor = Qfloor;
         let qceiling = Qceiling;
 
-        if !(round == qfloor || round == qceiling) {
```

Comment:
Sometimes it looks more readable one way, sometimes more readable the other way. It's like one of those pictures where the inverse of image is something else.

There's a Clippy check that we don't have enabled that checks for this. I submitted a PR related to it (https://github.com/rust-lang-nursery/rust-clippy/pull/3363), but to no avail so far.

Reply:
That PR talks about flipping the if/else over if one is too long or complicated so it reads better.

Here you reverted a DeMorgan. You went from `!(a || b)` to `!a && !b`. To my eye, the original code read better. I often read code out loud, much like an essay. `if round is not qfloor or qceiling` reads better to me than `if round is not floor and round is not ceiling`. Negations require more mental effort I think.

Reply:
IIRC, the complaint was about using a negated statement as the if-test. One way to get around that is to remove the negation and switch the then/else blocks. Another way, if possible, is to drive the negation inwards, which is what I did here. Generally I like my boolean expressions reduced, but since you feel strongly about it, I'll leave this one :)

(There are a lot of places in the code where we have negated if-tests, but they are all before then-blocks which are smaller than the else-blocks. It's annoying when Clippy complains about that, because that's clearly how it should be. That's what that PR is trying to fix.)

I'm also taking out the removal of a cast which I'm not sure about.

Reply:
Yeah, we are in bike-shedding territory here for sure. I find `!(a || b)` to be the "reduced" version of `!a && !b`. The `!` is a modifier. The first version has one modifier and the second has two. And as you mention, if you end up flipping the if/else over, it is easy to drop the `!` where as we need to apply an operation to the other. Sometimes the "inwards" shift makes the code more legible. It is always subjective.

I like your Clippy PR. It is a shame it is stuck.